pkg/server/server_controller_http.go

```diff
@@ -185,6 +186,12 @@ func (c *serverController) attemptLoginToAllTenants() http.Handler {
 					name:      string(name),
 					setCookie: setCookieHeader,
 				})
+				// In the case of /demologin, we want to redirect to the home page.
+				// If we get back a cookie along with a 302, be sure to transfer that
+				// into the ResponseWriter later.
+				if sw.code == 302 {
```

Comment:
`http.StatusTemporaryRedirect`?

Reply:
`http.StatusTemporaryRedirect` is assigned to code `307`. Seems like that's what we should be using anyway, so I went ahead and switched both places (the `/demologin` handler, and the quoted code above) to use `http.StatusTemporaryRedirect`.